tests/ssg_test_suite/rule.py

```diff
@@ -134,7 +151,12 @@ def perform_rule_check(options):
     ssg_test_suite.virt.start_domain(dom)
     domain_ip = ssg_test_suite.virt.determine_ip(dom)
     scanned_something = False
-    for rule_dir, rule, scripts in iterate_over_rules():
+
```

Comment:
This function is becoming really complex, I would appreciate extracting some functionality outside of it, but it is not a merge blocker.